engine/src/flutter/shell/platform/android/test/io/flutter/embedding/engine/loader/FlutterLoaderTest.java

```diff
@@ -399,4 +405,151 @@ public void itSetsShaderInitModeFromMetaData() {
     List<String> arguments = Arrays.asList(shellArgsCaptor.getValue());
     assertTrue(arguments.contains(shaderModeArg));
   }
+
+  @Test
+  public void itSetsAotSharedLibraryNameIfPathIsInInternalStorage() throws IOException {
+    FlutterJNI mockFlutterJNI = mock(FlutterJNI.class);
+    FlutterLoader flutterLoader = spy(new FlutterLoader(mockFlutterJNI));
+    Context mockApplicationContext = mock(Context.class);
+    File internalStorageDir = ctx.getFilesDir();
+    Path internalStorageDirAsPathObj = internalStorageDir.toPath();
+
+    ctx.getApplicationInfo().nativeLibraryDir =
+        Paths.get("some", "path", "doesnt", "matter").toString();
+    assertFalse(flutterLoader.initialized());
+    flutterLoader.startInitialization(ctx);
+
+    // Test paths for library living within internal storage.
+    Path pathWithDirectInternalStoragePath = internalStorageDirAsPathObj.resolve("library.so");
+    Path pathWithNestedInternalStoragePath =
+        internalStorageDirAsPathObj.resolve(Paths.get("some", "directories", "library.so"));
+    Path pathWithIndirectInternalStoragePath1 =
+        internalStorageDirAsPathObj.resolve(Paths.get("someDirectory", "..", "library.so"));
+    Path pathWithIndirectInternalStoragePath2 =
+        internalStorageDirAsPathObj.resolve(
+            Paths.get("some", "directory", "..", "..", "library.so"));
+    Path pathWithIndirectInternalStoragePath3 =
+        internalStorageDirAsPathObj.resolve(Paths.get("some", "directory", "..", "library.so"));
+
+    Path[] pathsToTest = {
+      pathWithDirectInternalStoragePath,
+      pathWithNestedInternalStoragePath,
+      pathWithIndirectInternalStoragePath1,
+      pathWithIndirectInternalStoragePath2,
+      pathWithIndirectInternalStoragePath3
+    };
+    String aotSharedNameArgPrefix = "--aot-shared-library-name=";
+
+    for (Path testPath : pathsToTest) {
+      String path = testPath.toString();
+      String aotSharedLibraryNameArg = aotSharedNameArgPrefix + path;
+      String[] args = {aotSharedLibraryNameArg};
+      flutterLoader.ensureInitializationComplete(ctx, args);
+      shadowOf(getMainLooper()).idle();
+
+      ArgumentCaptor<String[]> shellArgsCaptor = ArgumentCaptor.forClass(String[].class);
+      verify(mockFlutterJNI)
+          .init(
+              eq(ctx),
+              shellArgsCaptor.capture(),
+              anyString(),
+              anyString(),
+              anyString(),
+              anyLong(),
+              anyInt());
+
+      List<String> actualArgs = Arrays.asList(shellArgsCaptor.getValue());
+
+      // This check works because the tests run in debug mode. If run in release (or JIT release)
+      // mode, actualArgs would contain the default arguments for AOT shared library name on top
+      // of aotSharedLibraryNameArg.
+      String canonicalTestPath = testPath.toFile().getCanonicalPath();
```

Comment:
I am not sure which of the paths in this tests are just paths vs what files are created. Consider using [deleteOnExit](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#deleteOnExit()) for any files created during the test. 

As a general request can you run only this test file and see if any new files are created after the test is done?

Reply:
No files are created in these tests so I don't think deleteOnExit is necessary. The symlink tests I added, however, do create files so I made sure those are deleted at the end of the test!